src/main/java/org/icgc/argo/program_service/services/ProgramServiceFacade.java

```diff
@@ -106,7 +106,34 @@ public CreateProgramResponse createProgram(CreateProgramRequest request) {
             program,
             (ProgramEntity pe) -> {
               initializeProgramInEgo(pe, admins);
-            });
+            },
+            null);
+    log.debug("Created {}", programEntity.getShortName());
+    return programConverter.programEntityToCreateProgramResponse(programEntity);
+  }
+
+  @Transactional
+  public CreateProgramResponse createProgram(CreateProgramRequest request, UUID dataCenterId) {
+    val errors = validationService.validateCreateProgramRequest(request);
+    if (errors.size() != 0) {
+      throw Status.INVALID_ARGUMENT
+          .augmentDescription(
+              format("Cannot create program: Program errors are [%s]", join(errors, ",")))
+          .asRuntimeException();
+    }
+
+    val program = request.getProgram();
+    val admins = request.getAdminsList();
+
+    // TODO: Refactor this, having a transactional side effect is no longer needed thanks to the

```

Comment:
What is involved in refactoring this? I would prefer to build this right while we are looking at it instead of hoping someone addresses this in the future.